### DISCUSSION TITLE
Add watchOS support to snapshot

### DISCUSSION THREAD

**Discussion by ZevEisenberg** [fastlane/fastlane#18370] [Ideas]
Xcode 12.5 is adding support for running UI tests on watchOS. Snapshot should be updated to support screenshots on watchOS as well.

Some considerations:

- `fixLandscapeOrientation` in SnapshotHelper.swift uses `UIGraphicsImageRenderer`, which is not available on watchOS. Should this function be updated to use older CGContext-based rendering, or does this method not need to do anything special on watchOS?
- What about the simulator list in the Snapfile? Does it need separate lists of iOS and watchOS simulators?
- What about the scheme specifier in the Snapfile? Does it need separate schemes for iOS and watchOS?

**Comment by freak4pc** [comment]
I'd say +1 in general but given the smaller amount of watchOS consumers, should be prioritized accordingly

**Reply by joshdholtz** [reply]
I love this idea! Let's do it 😉

**Reply by mflint** [reply]
@joshdholtz I made a PR for this. https://github.com/fastlane/fastlane/pull/18890

**Comment by mflint** [comment]
I hacked together some changes last night to achieve this, so it is possible.

Will try to formalise this into a PR sometime in the next couple of weeks.

**Comment by Brett-Best** [comment]
I’ve also hacked something together but don’t have the knowledge to formalise it into a PR. https://github.com/Brett-Best/fastlane/tree/feature/snapshot-watchos-support

**Reply by mflint** [reply]
Very similar, @Brett-Best!

I made my PR - hoping one of the maintainers sees it: https://github.com/fastlane/fastlane/pull/18890
